### PR TITLE
RFC: New Code of Conduct and Moderation Policy

### PR DESCRIPTION
**Pull request by zkat**
##### Checklist

- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

- `rfcs`
- `policy`

##### Description

* [Rendered CoC](https://github.com/ayojs/ayo/blob/zkat/coc/CODE_OF_CONDUCT.md)
* [Rendered RFC](https://github.com/ayojs/ayo/blob/zkat/coc/rfcs/new-coc-and-moderation.md)

Hey all! Now that we have an actual @ayojs/moderation team, it's a good time to see about officially establishing our own CoC and moderation policies. So far, we've been vaguely and implicitly using the [Node.js Code of Conduct](https://github.com/nodejs/TSC/blob/master/CODE_OF_CONDUCT.md) for any moderation needs.

In essence, this RFC is proposing we move to a new CoC based on the [WeAllJS Code of Conduct](https://wealljs.org/code-of-conduct), which has its own built-in enforcement process and a few additions on top of the Contributor Covenant, which is what Node.js uses right now. Additionally, I'm proposing adapting a few bits from Node's current moderation policy 'cause they're nice and make sense, but I'm otherwise chucking their entire [Moderation Policy](https://github.com/nodejs/TSC/blob/master/Moderation-Policy.md) in the bin in favor of a more streamlined process that doesn't focus so much on punishment + reducing potential damage from punishment, which is how I read the current one.

I've included a **draft** of the CoC already adapted to use in this project. There's also a few bits missing that I haven't done yet that would be done after ratifying or before, since I don't know specifically what our reporting channels would be. This PR is a good place to discuss the specifics of those.

### CODE REVIEW

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: I know it's under construction. Just point some typo which obvious to me.

"it was really confusing" -> "it was really ridiculous" (or changing the above sentence)

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: `KeeperOfCommitBits` should be the `TheModerator`

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: `@keeperofbits` should be the `@TheModerator`

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: Ditto.

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: `KeeperOfCommitBits` should be the `TheModerator`

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [inline CODE_OF_CONDUCT.md]: Ditto.

File: CODE_OF_CONDUCT.md
Comment (Harrison-M) [reply CODE_OF_CONDUCT.md]: I like this as it stands. It demonstrates that you don't have to take the edit request with its exact wording, just follow the spirit and address the problem.

File: rfcs/new-coc-and-moderation.md
Comment (Harrison-M) [inline rfcs/new-coc-and-moderation.md]: We may want to replace or remove paranoia here. I know there's a push to try and limit its use as a descriptive term for mental health, and I think we'd prefer to err on the side of caution here, particularly if we can still get the point across with better wording.

File: rfcs/new-coc-and-moderation.md
Comment (Harrison-M) [inline rfcs/new-coc-and-moderation.md:92]: Is there a future possibility here for a team working to block a policy change they don't like, particularly if it's made in response to a team's behavior? Or would the moderation team's authority kick in prior to this?

File: rfcs/new-coc-and-moderation.md
Comment (zkat) [reply rfcs/new-coc-and-moderation.md]: ah! yes, I agree. :) Thank you!

File: rfcs/new-coc-and-moderation.md
Comment (zkat) [reply rfcs/new-coc-and-moderation.md:92]: Right now, since this is about inter-Team disagreements, I believe this may involve spawning off an [EFA team](https://github.com/ayojs/ayo/blob/latest/GOVERNANCE.md#elected-final-arbiters). I'm not sure whether to specify that as the process or leave it to the default?

File: CODE_OF_CONDUCT.md
Comment (sandfox) [inline CODE_OF_CONDUCT.md]: is it worth adding something along the lines of "(if possible) let them know why or how their actions/behaviour were inappropriate" ?
Good faith actors will appreciate the clarity and hopefully make it easier for them to do the "right thing" in future, and it gives bad faith actors one less step manoeuvre in.

File: CODE_OF_CONDUCT.md
Comment (sandfox) [inline CODE_OF_CONDUCT.md]: > doesn't have to do with

maybe replace with something like:

> doesn't have anything to do with

or 

> isn't directly related to?

to me at least the sentence reads very differently without the "anything"

**Review by varjmes — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by AgentAntelope — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (SomeHats) [reply CODE_OF_CONDUCT.md]: @sandfox I'm wary of making seem like it's the person on the receiving end who's responsible for explaining and justifying that in the moment. In my experience some sort of explanation tends to happen, but it might come a while after the incident and from someone else

**Review by SomeHats — APPROVED** [review]

**Review by stefee — APPROVED** [review]

File: CODE_OF_CONDUCT.md
Comment (addaleax) [inline CODE_OF_CONDUCT.md]: Can we have an explicit option of contacting one trusted member of the moderation team privately?

File: CODE_OF_CONDUCT.md
Comment (sandfox) [reply CODE_OF_CONDUCT.md]: @SomeHats completely agree that they shouldn't have to be responsible for justifying their hurt as in some cases it's just exhausting or causes further pain (by intent). I think I just want to try to nudge people towards communicating if they can/wish, but if you think this doesn't need spelling out so explicitly, or is going to used against it's spirit then quite happy to leave Kat's text as is.

File: CODE_OF_CONDUCT.md
Comment (ghost) [reply CODE_OF_CONDUCT.md]: i think this should be possible once #70 has landed?

File: rfcs/new-coc-and-moderation.md
Comment (scotttrinh) [reply rfcs/new-coc-and-moderation.md:92]: @Harrison-M 

We discussed this a lot on the Discord channel and we ended up leaving the language about how to spawn the EFA team intentionally vague to avoid unnecessary ceremony and overhead on getting up and running. We had two ideas for when consensus couldn't be met on an issue or RFC:

1. We elect a team of EFAs to serve for a predetermined term-length
2. We select a set of EFAs on an as-needed basis

Obviously neither of those solutions is fully fleshed out ("we" who?, "a set"? "team"? "term-length"?), but I think we're leaning toward the latter and we'll discover how that will come about in time, either through needing to make it happen, or having the bandwidth to give it a proper RFC.

File: CODE_OF_CONDUCT.md
Comment (NoahDragon) [reply CODE_OF_CONDUCT.md]: @Harrison-M Agree. Shouldn't we also point it out to the sample, or just imply it?

File: CODE_OF_CONDUCT.md
Comment (zkat) [reply CODE_OF_CONDUCT.md]: I'd like this, yes.

File: rfcs/new-coc-and-moderation.md
Comment (zkat) [reply rfcs/new-coc-and-moderation.md:92]: Anyone opposed to me leaving the language about this in _this_ RFC relatively vague and letting a different RFC go into specifics?

**Review by Qard — APPROVED** [review]

**Comment by Fishrock123** [comment]
seems reasonable to me

**Review by Harrison-M — APPROVED** [review]

**Review by NoahDragon — APPROVED** [review]

**Comment by ghost** [comment]
landed in db6ac6b!

**Pull request closed without merging**
